test/jdk/java/awt/im/4959409/bug4959409.java

```diff
@@ -22,51 +22,180 @@
+        robot.delay(DELAY);
+        robot.keyRelease(KeyEvent.VK_SHIFT);
+        robot.delay(DELAY);
+        robot.keyRelease(KeyEvent.VK_1);
+        robot.delay(DELAY);
+        robot.waitForIdle();
+
+        if (!keyPressedEventLatch.await(TIMEOUT, TimeUnit.SECONDS)) {
+            throw new RuntimeException("Did not received KEY PRESS for Shift + 1 , test failed");
+        }
+        return frame;
     }
-}
 
-class TestFrame extends JFrame implements KeyListener {
-    JTextField text;
-    JLabel label;
-
-    TestFrame () {
-        text = new JTextField();
-        text.addKeyListener(this);
-        label = new JLabel(" ");
-        Container c = getContentPane();
-        BorderLayout borderLayout1 = new BorderLayout();
-        c.setLayout(borderLayout1);
-        c.add(text, BorderLayout.CENTER);
-        c.add(label, BorderLayout.SOUTH);
-        setSize(300, 200);
-        setVisible(true);
+    public static void performMouseAction(final Robot robot, final int X, final int Y) {
+        robot.waitForIdle();
+        robot.delay(DELAY);
+        robot.mouseMove(X, Y);
+        robot.delay(DELAY);
+        robot.waitForIdle();
+
+        robot.delay(DELAY);
+        robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+        robot.delay(DELAY);
+        robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+        robot.delay(DELAY);
+        robot.waitForIdle();
     }
 
-    public void keyPressed(KeyEvent e) {
-        int code = e.getKeyCode();
-        int mods = e.getModifiers();
-        if (code == '1' && mods == KeyEvent.SHIFT_MASK) {
-            label.setText("KEYPRESS received for Shift+1");
+    public static boolean isComponentVisible(java.awt.Component component, CountDownLatch componentVisibleLatch) throws Exception {
+        int count = 1;
+        if (!component.isVisible()) {
+            while (count <= 5) {
+                TimeUnit.SECONDS.sleep(1);
+                if (component.isVisible()) {
+                    componentVisibleLatch.countDown();
+                    System.out.println("Component is visible " + component.toString());
+                    break;
+                }
+                count++;
+            }
         } else {
-            label.setText(" ");
+            componentVisibleLatch.countDown();
+            System.out.println("Component is visible " + component.toString());
         }
+        return componentVisibleLatch.await(TIMEOUT, TimeUnit.SECONDS);
     }
 
-    public void keyTyped(KeyEvent e) {
-    }
-
-    public void keyReleased(KeyEvent e) {
+    public static void main(String[] args) throws Exception {
+        bug4959409 bug4959409 = new bug4959409();
+        final JFrame[] jFrames = new JFrame[1];
+        try {
+            jFrames[0] = bug4959409.createUIAndTest();
+        } finally {
+            if (jFrames[0] != null) {
+                SwingUtilities.invokeAndWait(()-> {
+                    jFrames[0].dispose();
+                });
+            }
+        }
```

Comment:
Probably this could be simplified to:
```suggestion
        try {
            createUIAndTest();
        } finally {
            SwingUtilities.invokeAndWait(()-> {
                if (frame != null) {
                    frame.dispose();
                }
            });
        }
```

With the assumption `createUIAndTest()` is made static.

Reply:
done